ondemand.osc.edu/apps/myjobs/templates/Basic_NAMD_Parallel_Job_Pitzer/pbs_script.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+#PBS -N ondemand/sys/myjobs/basic_namd_parallel
+#PBS -l nodes=2:ppn=40
+#PBS -l walltime=0:12:00
+#PBS -S /bin/bash
+#PBS -j oe
+
+#   A Basic NAMD Parallel Job for the OSC Pitzer Cluster
+#   https://www.osc.edu/resources/available_software/software_list/namd
+
+#
+# The following lines set up the NAMD environment
+#
+module load intel/18.0.3  
+module load mvapich2/2.3
+module load namd/2.13b2
+#
+# Move to the directory where the job was submitted
+#
+cd $PBS_O_WORKDIR
+pbsdcp -p apoa1.namd *.xplor apoa1.psf apoa1.pdb $TMPDIR
```

Comment:
I'm still getting `Charm++> level of thread support used: MPI_THREAD_SINGLE (desired: MPI_THREAD_SINGLE)` on both Pitzer and Owens, which is not what we want for a "parallel" job.

Reply:
LAMMPS:
 `User defined values for MV2_CPU_BINDING_POLICY and MV2_THREADS_PER_PROCESS will lead to oversubscription of the available CPUs` can be solved by adding  `export MV2_ENABLE_AFFINITY=0` after `export OMP_NUM_THREADS=56`. 
This program combines MPI and OpenMP. We need to disable processor affinity by setting the environment variable $MV2_ENABLE_AFFINITY to 0. 
```
//Will use 56 threads without oversubscription
LAMMPS (22 Aug 2018)
  using 56 OpenMP thread(s) per MPI task
```
Let me know if it dones't work for you. 

NAMD:
I don't think there's anything we can do to change Charm++ config and solve `Charm++> level of thread support used: MPI_THREAD_SINGLE (desired: MPI_THREAD_SINGLE)`. I tried the parallel example from `~srb/workshops/compchem/namd/*2.13b2mpi.pitzer.pbs` and still get MPI_THREAD_SINGLE warning. It can be a 'parallel' job because output says `Info: Charm++/Converse parallel runtime startup completed at 0.0786467 s` and the job runs in the parallel queue.